### PR TITLE
test: Move FP conversion tests to a separate file

### PR DESCRIPTION
**Pull request by chfast**
After #618, GCC with sanitizer instrumentation goes OOM when compiling this file.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -25,16 +25,18 @@ ColumnLimit:     100
 ConstructorInitializerIndentWidth: 2
 IncludeBlocks: Preserve
 IncludeCategories:
+  - Regex:           '^"test/.*'
```
Comment (gumb0) [inline .clang-format:28]: can we sorty these lines by priority?

File: test/unittests/execute_floating_point_test.hpp [hunk 2]
```diff
@@ -0,0 +1,316 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "instructions.hpp"
+#include "types.hpp"
+#include "test/utils/floating_point_utils.hpp"
+#include "test/utils/hex.hpp"
```
Comment (gumb0) [inline test/unittests/execute_floating_point_test.hpp:8]: was this on purpose? other unit test files use angle bracket for test utils
```suggestion.
#include <test/utils/floating_point_utils.hpp>
#include <test/utils/hex.hpp>
```

File: test/unittests/execute_floating_point_conversion_test.cpp [hunk 3]
```diff
@@ -0,0 +1,559 @@
...
+#include "parser.hpp"
+#include "trunc_boundaries.hpp"
+#include <test/utils/asserts.hpp>
+#include <test/utils/hex.hpp>
+#include <cmath>
+
+using namespace fizzy;
+using namespace fizzy::test;
+
+TEST(execute_floating_point, f64_promote_f32)
```
Comment (gumb0) [inline test/unittests/execute_floating_point_conversion_test.cpp:17]: Maybe make these `execute_floating_point_conversion` suite

File: .clang-format (hunk 1 above)
Comment (chfast) [reply .clang-format:28]: No, because they are first match.

File: test/unittests/execute_floating_point_test.hpp (hunk 2 above)
Comment (chfast) [reply test/unittests/execute_floating_point_test.hpp:8]: I believe the `""` version is "more correct". The `<>` additionally adds system-level dirs, what is not what we want.

This is the `gcc -v` output:
```
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/9/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
```

But I can remove this change. I was just curious how would that look like.

**Comment by chfast** [comment]
The relation to #618 is not important any more, but I still think this is useful because the `execute_floating_point_test.cpp` is one of the longest compiling file in the project.

**Review by gumb0 — APPROVED** [review]

**Review by axic — APPROVED** [review]
Assuming it was really just renaming the file and moving out a big block verbatim. Have not gone through the pain to verify it.

**Pull request merged**